docker-compose.yml

```diff
@@ -2,7 +2,7 @@ version: '3'
 services:
   boulder:
     # Should match one of the GO_DEV_VERSIONS in test/boulder-tools/tag_and_upload.sh.
-    image: &boulder_image letsencrypt/boulder-tools:${BOULDER_TOOLS_TAG:-go1.20.5_2023-07-11}
+    image: &boulder_image letsencrypt/boulder-tools:${BOULDER_TOOLS_TAG:-go1.20.5_2023-07-12}
```

Comment:
This should be `go1.20.6_2023-07-12`. I was bit by this too. https://github.com/letsencrypt/boulder/pull/6995

Reply:
In `test/boulder-tools/tag_and_upload.sh`, line 53 could be changed to the following which will use the first entry in the GO_DEV_VERSIONS array for local testing.
```
sed -E "s|BOULDER_TOOLS_TAG:-go([0-9.]+)_([0-9-]+)}$|BOULDER_TOOLS_TAG:-go${GO_DEV_VERSIONS[0]}_${DATESTAMP}}|g" ../../docker-compose.yml
```